src/unicode.cpp

```diff
@@ -283,4 +283,19 @@ namespace ada::unicode {
       }
   }
 
+  // This function attemps to convert an ASCII string to a lower-case version.
+  // Once the lower cased version has been materialized, we check for the presence
+  // of the substring 'xn-', if it is found (unlikely), we then call the expensive 'to_ascii'.
+  ada_really_inline bool to_lower_ascii_string(std::optional<std::string>& out, size_t first_percent) noexcept {
```

Comment:
I think it would be simpler
```suggestion
  ada_really_inline bool to_lower_ascii_string(std::string& out, size_t first_percent) noexcept {
```

The function does not make use of the fact that you have an optional.

Reply:
If I make this, I receive the following error, because to_ascii function that this function uses still receives a std::optional<std::string>&.

```
Cannot bind non-const lvalue reference of type 'std::optional<std::__cxx11::basic_string<char> >&' to an rvalue of type
```

Reply:
Code needs to be adapted.

Reply:
I created a PR to illustrate my concern.

Reply:
Committed and fixed the issue. Thanks